v2/api/insights/customizations/component_extensions.go

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) Microsoft Corporation.
+ * Licensed under the MIT license.
+ */
+
+package customizations
+
+import (
+	"context"
+
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/conversion"
+
+	insights "github.com/Azure/azure-service-operator/v2/api/insights/v1api20200202/storage"
+	"github.com/Azure/azure-service-operator/v2/internal/genericarmclient"
+	"github.com/Azure/azure-service-operator/v2/internal/util/to"
+
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime"
+	"github.com/Azure/azure-service-operator/v2/pkg/genruntime/secrets"
+)
+
+var _ genruntime.KubernetesExporter = &ComponentExtension{}
+
+func (ext *ComponentExtension) ExportKubernetesResources(
+	ctx context.Context,
+	obj genruntime.MetaObject,
+	armClient *genericarmclient.GenericClient,
+	log logr.Logger,
+) ([]client.Object, error) {
+	// This has to be the current hub storage version. It will need to be updated
+	// if the hub storage version changes.
+	typedObj, ok := obj.(*insights.Component)
+	if !ok {
+		return nil, errors.Errorf("cannot run on unknown resource type %T, expected *redis.Redis", obj)
+	}
+
+	// Type assert that we are the hub type. This will fail to compile if
+	// the hub type has been changed but this extension has not
+	var _ conversion.Hub = typedObj
+
+	// Get our secretes
+	exportedSecrets, err := secretsToWrite(typedObj)
+	if err != nil {
+		return nil, err
+	}
+
+	return secrets.SliceToClientObjectSlice(exportedSecrets), nil
+}
+
+func secretsToWrite(obj *insights.Component) ([]*v1.Secret, error) {
+	operatorSpecSecrets := obj.Spec.OperatorSpec.Secrets
+	if operatorSpecSecrets == nil {
+		return nil, errors.Errorf("unexpected nil operatorspec")
+	}
+
+	collector := secrets.NewCollector(obj.Namespace)
+	collector.AddValue(operatorSpecSecrets.HockeyAppToken, to.Value(obj.Status.HockeyAppToken))
```

Comment:
Not following what this is doing - if it's on the status why is it a secret?

If this is just about making it exportable to a secret/configmap, the stuff I'm doing related to Dynamic export should cover it?

Reply:
Removed.